M2/Macaulay2/m2/fold.m2

```diff
@@ -1,4 +1,7 @@
 --		Copyright 1993-1999 by Daniel R. Grayson
+-- TODO: reverse is not copy-free, optimize this!
+-- TODO: drop is not copy-free, optimize this!
+-- TODO: both can be solved using iterators
```

Comment:
When reading the source for a project, it's much easier to see the tasks that need to be done right there, rather than being forces to search through ~600 issues for relevant tasks. There's also an issue for iterators: #1904.

Reply:
No, this file really has nothing to do with iterators, and if you posted an issue, there could be a discussion about it, with differences of opinion presented.

Reply:
I'm not sure what you're asking me here exactly since I didn't say this is the one and only way to solve it, I just mentioned two problems and a possible solution, so anyone reviewing the code in the future doesn't have to spend as much time. The point of mentioning iterators is that it is a solution to the first two items listed. Does adding a few comments in the code hurt anything?

Reply:
If it's a good idea, no one will ever see it.

By the way, I don't see how iterators allow "reverse" to be implemented.

Reply:
Changed the comment to point to the issue instead.